src/routes/legacyRoutes.ts

```diff
@@ -207,7 +211,7 @@ const asRegisterDeviceQuery = asObject({
 
 const asRegisterDeviceRequest = asObject({
   appId: asString,
-  tokenId: asString, // Firebase device token
+  tokenId: asOptional(asString), // Firebase device token
```

Comment:
How is this now optional? I would think it's required.

Reply:
It is optional in the database, and we already have records where it is missing, so I cannot break the API.